SimulationAgent/DeviceConnection/DeviceConnectionActor.cs

```diff
@@ -97,6 +104,24 @@ public enum ActorEvents
         /// </summary>
         public DeviceModel.DeviceModelMessage Message { get; private set; }
 
+        /// <summary>
+        /// Failed device connections counter
+        /// </summary>
+        public long FailedDeviceConnectionsCount => this.failedDeviceConnectionsCount;
+
+        /// <summary>
+        /// Failed device twin updates counter
+        /// </summary>
+        public long FailedTwinUpdatesCount => this.failedTwinUpdatesCount;
+
+        /// <summary>
+        /// Simulation error counter in DeviceConnectionActor
+        /// </summary>
+        public long SimulationErrorsCount => this.failedRegistrationsCount +
+            this.failedFetchCount +
+            this.failedTwinUpdatesCount +
+            this.failedDeviceConnectionsCount;
```

Comment:
for all these accesses to the private fields, use the public getter if available, e.g.
```
        public long SimulationErrorsCount => this.failedRegistrationsCount +
            this.failedFetchCount +
            this.FailedTwinUpdatesCount +
            this.FailedDeviceConnectionsCount;
```